src/index.ts

```diff
@@ -431,26 +437,18 @@ export class Upload extends Pumpify {
     return res;
   }
 
-  private async getRequestStream(reqOpts: r.OptionsWithUrl):
-      Promise<r.Request> {
+  private async getRequestStream(reqOpts: GaxiosOptions):
```

Comment:
Can we rename this? I was kind of confused when I noticed it no longer returns a stream.

Reply:
👍 and I think there are some more things I can do to simplify it.